WordPress/WordPressTest/LikeUserHelperTests.swift

```diff
@@ -28,46 +29,49 @@ class LikeUserHelperTests: XCTestCase {
 
     func testNewLikeUserWithPreferredBlog() {
         let completionExpectation = expectation(description: "We expect the context to save successfully")
-        let context = TestContextManager.sharedInstance().mainContext
+        let contextManager = TestContextManager()
+        let context = contextManager.mainContext
 
         let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
         let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
         let likeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
         XCTAssertNotNil(likeUser)
 
-        TestContextManager.sharedInstance().save(context) {
+        XCTAssertNoThrow(contextManager.save(context, withCompletionBlock: {
             completionExpectation.fulfill()
-        }
+        }))
 
         waitForExpectations(timeout: 0.1)
     }
 
     func testUpdatingExistingUserToRemovePreferredBlog() {
         let completionExpectation = expectation(description: "We expect the context to save successfully")
-        let context = TestContextManager.sharedInstance().newDerivedContext()
+        let contextManager = TestContextManager()
+        let context = contextManager.mainContext
 
         // First we create the pre-existing user, so we can later modify it to remove the preferred blog
 
         let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
         let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
         let existingLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
-        let existingPreferredBlog = existingLikeUser.preferredBlog
-        XCTAssertNotNil(existingLikeUser)
-
-        TestContextManager.sharedInstance().save(context)
+        guard let existingPreferredBlog = existingLikeUser.preferredBlog else {
+            XCTFail()
+            return
+        }
 
-        //let updateContext = TestContextManager.sharedInstance().newDerivedContext()
+        XCTAssertNotNil(existingLikeUser)
+        XCTAssertNoThrow(contextManager.save(context))
 
         // Then we remove the preferred blog from the remote user, so we can save it again and make sure
         // the preferred blog deletion works fine.
         remoteUser.preferredBlog = nil
         let updatedLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
         XCTAssertNotNil(updatedLikeUser)
-        XCTAssertTrue(existingPreferredBlog?.isDeleted ?? false)
+        XCTAssertTrue(existingPreferredBlog.isDeleted)
 
-        TestContextManager.sharedInstance().save(context) {
+        XCTAssertNoThrow(contextManager.save(context, withCompletionBlock: {
             completionExpectation.fulfill()
-        }
+        }))
```

Comment:
I tested this by `git revert`-ing commit `7dca8d7fe4c8270e5c5f3a9cefc221f381b2d137` (which applies the fix for the issue), then running the `LikeUserHelperTests` test suite… and it seems I still got the test runner to crash (and not fail)… 😒 

```
*** Terminating app due to uncaught exception 'Unresolved Core Data save error (Main Context)', reason: 'Non-optional property with a nil value on LikeUserPreferredBlog.user'
```

So seems like `XCTAssertNoThrow` is not enough here to catch that `NSException`? (Maybe because the throw is happening async on a separate thread? 🤔 )

Reply:
Rolled back the changes to the initial version, as discussed.

Reply:
Ahah seems you commented here at the same time I added my review+comment and we crossed the streams 😅